lib/src/delegates/asset_picker_builder_delegate.dart

```diff
@@ -1231,21 +1233,25 @@ class DefaultAssetPickerBuilderDelegate
       builder: (context, wrapper, _) {
         // First, we need the count of the assets.
         int totalCount = wrapper?.assetCount ?? 0;
-        final Widget? specialItem;
-        // If user chose a special item's position, add 1 count.
-        if (specialItemPosition != SpecialItemPosition.none) {
-          specialItem = specialItemBuilder?.call(
-            context,
-            wrapper?.path,
-            totalCount,
-          );
-          if (specialItem != null) {
-            totalCount += 1;
-          }
-        } else {
-          specialItem = null;
-        }
-        if (totalCount == 0 && specialItem == null) {
+
+        final List<SpecialItemModel> specialItemModels = specialItems
+            .map((item) {
+              return (
+                position: item.position,
+                item: item.builder?.call(
+                  context,
+                  wrapper?.path,
+                  totalCount,
+                  permissionNotifier.value,
+                )
+              );
+            })
+            .whereType<SpecialItemModel>()
+            .toList();
+
+        totalCount += specialItemModels.length;
```

Comment:
Looks redundant.

Reply:
I think `totalCount += specialItemModels.length` is required here.

Reply:
Sorry, what I refer to is `whereType` is meaningless and the `+=` can rely on `specialItems.length`.

Reply:
`whereType` is required because the `item.builder?.call` might return null but `SpecialItemModel` required non-null widget. 

```dart
 final List<SpecialItemModel> specialItemModels = specialItems
            .map((item) {
              return (
                position: item.position,
                item: item.builder?.call(
                  context,
                  wrapper?.path,
                  totalCount,
                  permissionNotifier.value,
                )
              );
            })
            .whereType<SpecialItemModel>()
            .toList();
```

```dart
typedef SpecialItemModel = ({
  SpecialItemPosition position,
  Widget item,
});
```